domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementsController.cs

```diff
@@ -23,71 +22,89 @@ namespace API.Transfer.Api.Controllers;
-        this.createTransferAgreementValidator = createTransferAgreementValidator;
         this.projectOriginWalletService = projectOriginWalletService;
         this.httpContextAccessor = httpContextAccessor;
+        this.transferAgreementProposalRepository = transferAgreementProposalRepository;
     }
 
-    [ProducesResponseType(201)]
-    [ProducesResponseType(409)]
-    [ProducesResponseType(400)]
+    /// <summary>
+    /// Add a new Transfer Agreement
+    /// </summary>
+    /// <param name="request">The request object containing the TransferAgreementProposalId for creating the Transfer Agreement.</param>
+    /// <response code="201">Successful operation</response>
+    /// <response code="400">Only the receiver company can accept this Transfer Agreement Proposal or the proposal has run out</response>
+    /// <response code="404">TransferAgreementProposal expired or deleted</response>
+    /// <response code="409">There is already a Transfer Agreement with proposals company tin within the selected date range</response>
     [HttpPost]
-    public async Task<ActionResult> Create([FromBody] CreateTransferAgreement request)
+    [ProducesResponseType(typeof(TransferAgreement), 201)]
+    [ProducesResponseType(typeof(void), 400)]
+    [ProducesResponseType(typeof(void), 404)]
+    [ProducesResponseType(typeof(void), 409)]
+    public async Task<ActionResult> Create(CreateTransferAgreement request)
     {
-        var subject = User.FindSubjectGuidClaim();
-        var subjectName = User.FindSubjectNameClaim();
-        var subjectTin = User.FindSubjectTinClaim();
+        if (request.TransferAgreementProposalId == Guid.Empty)
+        {
+            return BadRequest("Must set TransferAgreementProposalId");
+        }
 
-        var validateResult = await createTransferAgreementValidator.ValidateAsync(request);
-        if (!validateResult.IsValid)
+        var proposal = await transferAgreementProposalRepository.GetNonExpiredTransferAgreementProposal(request.TransferAgreementProposalId);
+        if (proposal == null)
         {
-            validateResult.AddToModelState(ModelState);
-            return ValidationProblem(ModelState);
+            return NotFound("TransferAgreementProposal expired or deleted");
         }
 
-        var transferAgreement = new TransferAgreement
+        var subjectTin = User.FindSubjectTinClaim();
+        if (proposal.ReceiverCompanyTin != subjectTin)
         {
-            StartDate = DateTimeOffset.FromUnixTimeSeconds(request.StartDate),
-            EndDate = request.EndDate.HasValue ? DateTimeOffset.FromUnixTimeSeconds(request.EndDate.Value) : null,
-            SenderId = Guid.Parse(subject),
-            SenderName = subjectName,
-            SenderTin = subjectTin,
-            ReceiverTin = request.ReceiverTin
-        };
+            return BadRequest("Only the receiver company can accept this Transfer Agreement Proposal");
```

Comment:
According to our guidelines 400 Bad request must return a validation problem

Reply:
corrected